### PR TITLE
Issue #66 - Form makeover

### PR DESCRIPTION
**Pull request by EChesters**
@womenhackfornonprofits/whfnp

[Issue-77](https://github.com/womenhackfornonprofits/whfnp/issues/77)
-  button sizes reconstructured
-  Form classes to use ones from Bootstrap
-  Added new colour for help-text
-  Added accessibility required aria attributes
-  Increased font-weight so it's readable and changed colours to black rather than red

### CODE REVIEW

**Comment by EChesters** [comment]
_This is actually Issue #113 and not Issue 66_ I saw the wrong issue when I was typing the commit message and I can't change it.

**Comment by EChesters** [comment]
Volunteer contact form:
screen shot 2016-06-19 at 21 18 42

NP contact form:
screen shot 2016-06-19 at 21 17 59

Contact form:
screen shot 2016-06-19 at 20 03 28

**Comment by EChesters** [comment]
In terms of the names fields I have some questions:
1. Is there any reason why we separate them?
2. Will there be any back end issue not making surname a required field - not everyone in this world has a surname. 

_Can we remove the surname field and combine it with first name, or just remove the required attribute?_

**Comment by Nandhini31** [comment]
@EChesters :+1: Yes fine with all the new forms. happy to have just one name field.

File: src/contact.html [hunk 1]
```diff
@@ -7,20 +7,32 @@
...
-		<input type="email" name="_replyto" placeholder="E.g. jon.snow@example.com" required>
-		<input type="hidden" name="_subject" value="New Submission: WHFN Website Contact Form"/>
-		<input type="text" name="_gotcha" style="display:none"/>
-    <label for="tel">Phone Number:</label>
-		<input type="tel" name="phone" placeholder="E.g. 07912 345 678">
-    <label for="message">Enter your message here:</label>
-		<textarea name="message" rows="5" placeholder="E.g. Hello, just wanted to tell you I think you are awesome" required></textarea>
-		<input type="submit" value="Send">
+		<div class="form-group">
+	    	<label for="name">Name</label>
```
Comment (tanyapowell) [inline src/contact.html:14]: This is not specifically for this line, but generally for throughout this PR (wanted to use this line as an example though) 
- Wrap radio and checkbox inputs and their text in labels. No need for `for` attributes. The wrapping automatically associates the two.

```
<!-- bad -->
<label for="email">
<input id="email" name="email">

<!-- good -->
<label>Name: <input name="email"></label>
```

File: src/contact.html [hunk 2]
```diff
@@ -7,20 +7,32 @@
...
+			<input type="text" name="name" placeholder="Ada Lovelace" required aria-required="true" class="form-control"/>
+		</div>
+		<div class="form-group">
```
Comment (tanyapowell) [inline src/contact.html:17]: Really we should avoid the extra overhead of having the `label` and `input` tag inside a parent `div`. Is this a bootstap thing why each `label`/`input` tag inside a parent `div`?

File: src/sass/partials/_button.scss [hunk 3]
```diff
@@ -1,26 +1,34 @@
...
-		background: none;
-		color: $pop-pink;
+		background: darken($pop-pink, 10%);
+		text-decoration: underline;
+		border-color: darken($pop-pink, 15%);
+		color: $white;
 	}
 	
-}
-.larger-button {
```
Comment (tanyapowell) [inline src/sass/partials/_button.scss:19]: Has this removal affected other buttons on the site?

File: src/sass/partials/_colours.scss [hunk 4]
```diff
@@ -1,37 +1,40 @@
...
-$volunter-name-bg-color-2: rgba(247, 203, 255, 0.6);
-
-$grey-color:       #828282;
-$grey-color-light: lighten($grey-color, 30%);
-$grey-color-dark:  darken($grey-color, 25%);
-
-$brand-color:      #ff6525;
-$link-hover:	   darken($brand-color, 15%);
\ No newline at end of file
+$white: 						#fff;
```
Comment (tanyapowell) [inline src/sass/partials/_colours.scss:1]: 🙈 did you reformat this file? It's a bit hard to read now

**Comment by tanyapowell** [comment]
This is just my opinion, but it looks a little weird __only__ have the name field with the little icon on the right hand side of the `input` box. 

image

Could you add one to email too?

File: src/sass/partials/_form.scss [hunk 5]
```diff
@@ -1,104 +1,7 @@
...
-  padding: $form-padding-value;
-  margin: 0 auto $form-padding-value auto;
-  border: none;
-  color: $black;
-  background-color: #FDEFF0;
-  font-weight: 100;
-  box-shadow: 1px 1px 1px 1px rgba(0,0,0,.1) inset
-
-}
+span.help-text {
```
Comment (lili2311) [inline src/sass/partials/_form.scss:1]: This is too specific, please avoid selectors like these just .help-text is sufficient to select the correct element.

Comment (tanyapowell) [inline src/sass/partials/_form.scss:1]: Think bootstrap has a `.help-text`. So this is how your override it, by explicitly declaring it this way

Comment (lili2311) [inline src/sass/partials/_form.scss:1]: Since we import bootstrap first thing any styles you write afterwards naturally override it.

Comment (tanyapowell) [inline src/sass/partials/_form.scss:1]: From my experience this is not always the case. Bootstrap has an annoying habit of overriding your styling if it's not explicitly instructed not to. If someone knows a way around this I'd love to know or maybe this is fixed in V4

Comment (lili2311) [inline src/sass/partials/_form.scss:1]: We imported and included bootsrap sass manually though we are not using an external stylesheet. So we have a lot more flexibility and control over what we use from bootsrap.

**Comment by tanyapowell** [comment]
@Nandhini31 were you giving the 👍 on the whole pull request or just answering the question?

**Comment by EChesters** [comment]
@tanyapowell The icon in the field next to name is my LastPass auto fill plugin. That isn't something I've done and it'll go away on other people's screens. 

@lili2311 The selector wasn't that specific since help text when used on a site is used everywhere. Bootstrap also has this selector and I was overriding that. 

I'm just about to change the name fields to combine them and use @tanyapowell's way of wrapping the input inside label. Will the names being combined into one cause any issues when submitting and the way we expect the data?

File: src/sass/partials/_button.scss (hunk 3 above)
Comment (EChesters) [inline src/sass/partials/_button.scss:19]: I checked where this was being used and I didn't find any occurrences.

**Comment by lili2311** [comment]
`.larger-button` is being used in the projects page, please check it wont affect it

**Comment by EChesters** [comment]
I tried @tanyapowell's suggestion but it broke the styling. This has been sidelined for now by Tanya. 

Top of volunteers form now just says name. **Will this break submitting the form?**
screen shot 2016-06-23 at 23 20 39

File: src/contact.html [hunk 6]
```diff
@@ -7,20 +7,32 @@
...
 </section>
-<section>
-	<h2> Get in touch </h2>
+<section class="col-lg-3 col-md-7 col-sm-9 col-xs-12 position central">
+	<h2>Get in touch</h2>
 	<form method="POST" action="//formspree.io/help@womenhackfornonprofits.com">
-    <label for="name">Name:</label>
-		<input type="text" name="name" placeholder="E.g. Jon Snow" required>
-    <label for="email">Email Address:</label>
-		<input type="email" name="_replyto" placeholder="E.g. jon.snow@example.com" required>
```
Comment (lili2311) [inline src/contact.html:16]: `name="_replyto"` is required to make the form submission work, same for other fields that have `name="_smth"` please test the form locally perhaps?  just to make sure it still works, we will need to approve the email again so ti submits locally.

File: src/non-profits-contact.html [hunk 7]
```diff
@@ -2,31 +2,45 @@
...
 title: Non-Profits Contact | Women Hack for Non-Profits
 ---
-
 <section class="smaller-banner">
   <div class="content">
     <h1 class="title-heading">Non-Profits Contact Form</h1>
   </div>
 </section>
-<section>
+<section class="col-lg-3 col-md-7 col-sm-9 col-xs-12 position central">
```
Comment (lili2311) [inline src/non-profits-contact.html]: is there a screenshot perhaps of what it looks like now under different break points?

File: src/contact.html (hunk 6 above)
Comment (EChesters) [inline src/contact.html:16]: The `_replyto` hasn't been touched, and is still part of the email line.

**Comment by EChesters** [comment]
`.col-sm-9`
screen shot 2016-06-24 at 19 53 05

`.col-md-7`
screen shot 2016-06-24 at 19 53 15

`.col-xs-12`
screen shot 2016-06-24 at 19 53 19

_Please see the above for `.col-lg-3`!!_

**Comment by lili2311** [comment]
.col-sm-9 and .col-sm-7 both look a little too wide in my opinion, perhaps smaller width?

**Comment by tanyapowell** [comment]
@EChesters can you send over your wireframes for any forms? If the width in your screengrab matches (or is near enough to the wireframe) then it'll be fine 😄

**Comment by EChesters** [comment]
The prototype is here: http://echesters.co.uk/WHFNP-Website/ but I did give up on them half way through and didn't get round to doing the forms. On the wireframes these also weren't done because I hadn't done the contact pages.

**Comment by tanyapowell** [comment]
Just been through the code and it's all good with me 👍🏿

On Wed, 6 Jul 2016 at 14:40 Elizabeth Chesters notifications@github.com
wrote:

> The prototype is here: http://echesters.co.uk/WHFNP-Website/ but I did
> give up on them half way through and didn't get round to doing the forms.
> On the wireframes these also weren't done because I hadn't done the contact
> pages.
> 
> —
> You are receiving this because you were mentioned.
> 
> Reply to this email directly, view it on GitHub
> https://github.com/womenhackfornonprofits/whfnp/pull/113#issuecomment-230773973,
> or mute the thread
> https://github.com/notifications/unsubscribe/AKlONUycthCrhgHU5IB3JhztbyG3RrdVks5qS7A3gaJpZM4I5On6
> .
> 
> ## 

Tanya Powell
about.me/tanyapowell
https://about.me/tanyapowell?promo=email_sig

File: src/contact.html (hunk 1 above)
Comment (EChesters) [inline src/contact.html:14]: Tried this, and it broke Bootstrap's styling. This was decided with @tanyapowell to try at a later stage.

File: src/contact.html (hunk 2 above)
Comment (EChesters) [inline src/contact.html:17]: This is for Bootstrap styling, although I agree with you. I'm not sure why Bootstrap do this. See on here for their example: http://getbootstrap.com/css/#forms

File: src/sass/partials/_colours.scss (hunk 4 above)
Comment (EChesters) [inline src/sass/partials/_colours.scss:1]: I'll clean this up. For some reason it's all over the place on here but not on my Sublime.

**Pull request merged**